### ISSUE TITLE
Can't launch apps on Windows 10

### ISSUE THREAD

**Issue by ptmkenny** [bayleeadamoss/zazu#134]
Related issue: 
https://github.com/tinytacoteam/zazu/issues/118
(for Linux, this is for Windows)

### Expected Behavior

When I type Chrome or Word,
When I type Chrome or Word, I can launch the app by pressing enter.

### Actual Behavior

I get a prompt to search Google for Chrome or Word.

### Steps to reproduce the behavior

Type the name of an app.

### Versions

* Zazu: 0.3.2
* OS: Windows 10
* GIT (if you have it installed):

**Comment by bayleedev** [comment]
@ptmkenny does this file exist? `C:\Users\<YOUR_USER>\.zazu\plugins\tinytacoteam\zazu-file-finder\data\applications.json` ?

**Comment by ptmkenny** [comment]
In C:\Users\<YOUR_USER>\.zazu\plugins\tinytacoteam\zazu-file-finder\data\ the only file is .gitkeep, 0KB.

**Comment by bayleedev** [comment]
What's inside of `C:\Users<YOUR_USER>.zazu\plugins\tinytacoteam\zazu-file-finder\node_modules` ?

**Comment by ptmkenny** [comment]
There's a plist folder, which in turn contains:

dist/
examples/
lib/
.jshintrc
.travis.yml
History.md
LICENSE
Makefile
README.md
index.js
package.json

On Tue, Jan 10, 2017 at 2:18 PM, Blaine Schmeisser <notifications@github.com

**Comment by YonghoChoi** [comment]
I have the same problem on windows 10.

**Comment by drazik** [comment]
Hello,

Same problem here. The only difference is that `C:\Users<YOUR_USER>.zazu\plugins\tinytacoteam\zazu-file-finder\node_modules` contains the following directories / files on my machine : 

base64-js
plist
xmlbuilder
xmldom

**Comment by scheduardo** [comment]
Same here. Nothing happens!

Windows 10!

See: https://cl.ly/j2Xy

**Comment by twang2218** [comment]
If your system has `node` installed, could you try the following command under `C:\Users\<YOUR_USER>\.zazu\plugins\tinytacoteam\zazu-file-finder\` directory, and paste the result here?

```cmd
cd C:\Users\<YOUR_USER>\.zazu\plugins\tinytacoteam\zazu-file-finder\
node appCache.js '{}'
```

This will try to manually call the `data\application.json` generator, if any exception happened there, it will be output to the console.

**Comment by TopGrd** [comment]
I have same problem in my os: Windows 10. When i check the zazu-file-finder folder, i can't find application.json, so i run your command  
```js
cd C:\Users\<YOUR_USER>\.zazu\plugins\tinytacoteam\zazu-file-finder\
node appCache.js '{}'
```
some modules is not found like follow picture.
![](http://i1.piimg.com/567571/63aed1f5f5050ee6.png)  
I run command `npm install && node appCache.js {}`, then SUCCESS!!
![](http://i1.piimg.com/567571/72ec3da9a8414915.png)

**Comment by twang2218** [comment]
@TopGrd `async` is newly added dependencies in `zazu-file-finder`, it seems the plugin update process failed to install the new dependencies somehow. So, the zazu file/app finder works now?

**Comment by TopGrd** [comment]
@twang2218 not only `async` modules, like follow pictures.  
![](http://p1.bqimg.com/567571/3e01a020abfe57ac.png)  
![](http://p1.bqimg.com/567571/0ee1b26ade72092a.png) 

Now the zazu file/app finder works well.

**Comment by bayleedev** [comment]
@TopGrd do you still have the log file from the day you upgraded?

`~/.zazu/logs`

**Comment by TopGrd** [comment]
@blainesch  Yes, it's in my computer. There is download link if you  need it.
 [log](https://www.dropbox.com/s/6xdp97172f21hqs/zazu.log.2017-02-09?dl=0)

**Comment by twang2218** [comment]
@TopGrd I didn't see the `zazu-file-finder` updating log, here is what I saw:

```bash
 cat zazu.log.2017-02-09| grep 'npm'
{"plugin":"tinytacoteam/zazu-calculator","level":"info","message":"npm install"}
{"level":"verbose","message":"npm install [tinytacoteam/zazu-calculator] has 3 attempts"}
{"plugin":"tinytacoteam/zazu-system","level":"info","message":"npm install"}
{"level":"verbose","message":"npm install [tinytacoteam/zazu-system] has 3 attempts"}
{"plugin":"tinytacoteam/zazu-template","level":"info","message":"npm install"}
{"level":"verbose","message":"npm install [tinytacoteam/zazu-template] has 3 attempts"}
{"level":"verbose","message":"npm install [tinytacoteam/zazu-system] success"}
{"plugin":"blakek/zazu-ip-address","level":"info","message":"npm install"}
{"level":"verbose","message":"npm install [blakek/zazu-ip-address] has 3 attempts"}
{"level":"verbose","message":"npm install [tinytacoteam/zazu-template] success"}
{"plugin":"tinytacoteam/zazu-clipboard","level":"info","message":"npm install"}
{"level":"verbose","message":"npm install [tinytacoteam/zazu-clipboard] has 3 attempts"}
```

And I didn't see the npm failures either:

```bash
 cat zazu.log.2017-02-09| grep failed
{"error":{},"plugin":"blakek/zazu-ip-address","block":"ciyxqkbq70000ucot7ml2rf5m","level":"error","message":"Script failed"}
{"error":{},"plugin":"blakek/zazu-ip-address","block":"ciyxqkbq70000ucot7ml2rf5m","level":"error","message":"Script failed"}
```

Did you update the `zazu-file-finder` at `2017-02-09`? or another day? Could you run `grep npm zazu.log.*` at the log directory, maybe it's in another file.

Just based on this file, I found something interesting, there are **5** `npm install` request, but only **2** success without any failures. And without the `zazu is closing` message, I saw `app is ready`, which means you restart the Zazu(or system) during the installation process, which caused the installation interruption and might be the cause of dependencies broken.

If that is the case, run the `Update plugins` again might fix the problem, otherwise, you may need manually run `npm install` for each plugin (or even remove the `node_modules` directory under each plugin and do a clean `npm install` again).

**Comment by TopGrd** [comment]
@twang2218  Ok, i just installed zazu application at 2017-02-09, not update. Maybe it's my fault to interrupt the dependencies installation. i forget my actions during install.

**Comment by drazik** [comment]
Hello. If it can help, I have just updated Zazu (2017-02-10) today and the issue is still present.

I tried [this](https://github.com/tinytacoteam/zazu/issues/134#issuecomment-277491561) and get no error.

**Comment by twang2218** [comment]
@drazik Is there a `C:\Users<YOUR_USER>.zazu\plugins\tinytacoteam\zazu-file-finder\data\application.json` file? If you got no error of `node appCache.js '{}'`, then probably it's working properly. Could you provide more information about your problem? Thanks.

**Comment by drazik** [comment]
Woops, I think I just needed to wait for indexing after the update, because now it's working properly. I'm sorry to have disturbed you for nothing. It seems the recent update fixed the problem. Good job, thank you very much 👍

**Comment by ohardy** [comment]
Same bug on MacOs with master branch when running node appCache.js '{}'

```
Error: Cannot find module 'app2png'
```

**Comment by twang2218** [comment]
@ohardy Could you try `npm install --production`, it seems missing the dependencies somehow.

**Comment by ohardy** [comment]
@twang2218 Yep I did before you ask because I know. But it's look like à bug in Zazu that sometime failed to do npm install after plugin update

**Comment by twang2218** [comment]
@ohardy There is a PR for sub-dependencies issue, https://github.com/tinytacoteam/zazu/pull/252, but `app2png` is direct dependency, do you have the log of the day you update the plugin? Could you post the log here if you still have it? I think there might be some error in the log.
